src/main/java/nitezh/ministock/StockSuggestions.java

```diff
@@ -46,43 +46,76 @@ of this software and associated documentation files (the "Software"), to deal
+            Cache stockCache = new StorageCache(null);
+            stockResponse = UrlDataTools.getCachedUrlData(stockUrl, stockCache, 86400);
 
         } catch (UnsupportedEncodingException e1) {
-            response = null;
+            e1.printStackTrace();
+            stockResponse = null;
         }
 
         // Return if empty response
-        if (response == null || response.equals("")) {
+        if (stockResponse == null || stockResponse.equals("") ||
+                cryptoResponse == null || cryptoResponse.equals("")) {
             return suggestions;
         }
-        Matcher m = PATTERN_RESPONSE.matcher(response);
-        if (m.find()) {
-            response = m.group(1);
-            try {
-                JSONArray jsonA = new JSONObject(response)
+
+        try {
+            JSONArray cryptoJson = new JSONArray(cryptoResponse);
+
+            for (int i = 0; i < cryptoJson.length(); i++) {
+
+                String symbol = cryptoJson.getJSONObject(i).getString("symbol");
+
+                Matcher cryptoMatcher = STOCK_REGEX.matcher(symbol);
+                if (cryptoMatcher.matches() && query.length() > 0) {
+                    Map<String, String> suggestion = new HashMap<>();
+                    suggestion.put("symbol", symbol);
+                    suggestion.put("name", cryptoJson.getJSONObject(i).getString("name"));
+                    suggestions.add(suggestion);
+                }
+            }
+
+            Matcher m = PATTERN_RESPONSE.matcher(stockResponse);
+            if (m.find()) {
+                stockResponse = m.group(1);
+
+                JSONArray jsonA = new JSONObject(stockResponse)
                         .getJSONObject("ResultSet")
                         .getJSONArray("Result");
 
                 for (int i = 0; i < jsonA.length(); i++) {
-                    Map<String, String> suggestion = new HashMap<>();
                     JSONObject jsonO = jsonA.getJSONObject(i);
-                    suggestion.put("symbol", jsonO.getString("symbol"));
-                    suggestion.put("name", jsonO.getString("name"));
-                    suggestions.add(suggestion);
-                }
-                return suggestions;
 
-            } catch (JSONException ignored) {
+                    String symbol = jsonO.getString("symbol");
+                    Matcher stockMatcher = STOCK_REGEX.matcher(symbol);
+
+                    // 7 characters is enough for StockSymbol.CountryCode, ex: AAPL.MX
+                    if (stockMatcher.matches() && symbol.length() < 8) {
+                        Map<String, String> suggestion = new HashMap<>();
+                        suggestion.put("symbol", symbol);
+                        suggestion.put("name", jsonO.getString("name"));
+                        suggestions.add(suggestion);
+                    }
+                }
             }
+            return suggestions;
+        } catch (JSONException ignored) {
```

Comment:
Is nothing supposed to happen when the exception is caught?

Reply:
its needed for JSONArray